workflow/rules/long_read.smk

```diff
@@ -119,40 +97,50 @@ rule canu_correct:
-        " corOverlapper=minimap utgOverlapper=minimap obtOverlapper=minimap"
-        " corConcurrency={params.concurrency}"
-        " cormhapConcurrency={params.concurrency} cormhapThreads={params.concurrency}"
-        " cormmapConcurrency={params.concurrency} cormmapThreads={params.concurrency}"
-        " obtmmapConcurrency={params.concurrency} obtmmapThreads={params.concurrency}"
-        " utgmmapConcurrency={params.concurrency} utgmmapThreads={params.concurrency}"
-        " redConcurrency={params.concurrency} redThreads={params.concurrency}"
-        " ovbConcurrency={params.concurrency}"
-        " ovsConcurrency={params.concurrency}"
-        " oeaConcurrency={params.concurrency})"
-        "> {log} 2>&1"
-
-
-# rule medaka_consensus_reference:
-use rule assembly_polishing_ont as medaka_consensus_reference with:
+        """
+        ( if [ -d {params.outdir} ]; then rm -Rf {params.outdir}; fi &&
+        canu -correct -nanopore {input} -p {wildcards.sample} -d {params.outdir} genomeSize=30k minOverlapLength=10 minReadLength=200 \
+        useGrid=false {params.for_testing} \
+        corMMapMerSize=10 corOutCoverage=50000 corMinCoverage=0 maxInputCoverage=20000 \
+        corOverlapper=minimap utgOverlapper=minimap obtOverlapper=minimap \
+        corConcurrency={params.concurrency} \
+        cormhapConcurrency={params.concurrency} cormhapThreads={params.concurrency} \
+        cormmapConcurrency={params.concurrency} cormmapThreads={params.concurrency} \
+        obtmmapConcurrency={params.concurrency} obtmmapThreads={params.concurrency} \
+        utgmmapConcurrency={params.concurrency} utgmmapThreads={params.concurrency} \
+        redConcurrency={params.concurrency} redThreads={params.concurrency} \
+        ovbConcurrency={params.concurrency} \
+        ovsConcurrency={params.concurrency} \
+        oeaConcurrency={params.concurrency})
+        gzip -d {output.corr_gz}
+        2> {log}
+        """
+
+
+rule clip_adbc_corrected:
     input:
-        fasta="results/{date}/corrected/{sample}/{sample}.correctedReads.fasta.gz",
-        reference="resources/genomes/main.fasta",
+        primer=get_artic_primer,
+        reads="results/{date}/corrected/{sample}/{sample}.correctedReads.fasta",
+        ref_genome="resources/genomes/main.fasta",
     output:
-        temp("results/{date}/consensus/medaka/{sample}/{sample}.fasta"),
+        "results/{date}/norm_trim_corr_reads/{sample}/{sample}.clip.fasta",
+    params:
+        outdir="results/{date}/norm_trim_corr_reads/{sample}",
+    log:
+        "results/{date}/norm_trim_corr_reads/{sample}/notramp.log",
+    conda:
+        "../envs/notramp.yaml"
+    shell:
+        "notramp -t --incl_prim -r {input.reads} -p {input.primer} -g {input.ref_genome} -o {params.outdir}"
 
 
-# polish consensus
 rule bcftools_consensus_ont:
     input:
-        fasta="results/{date}/consensus/medaka/{sample}/{sample}.fasta",
+        fasta="results/{date}/consensus/medaka/{sample}/consensus.fasta",
```

Comment:
maybe a simpler folder structure would help here, samplename included in filename

Reply:
Let's discuss with Thomas for this particular rule, because he implemented it afaik (unless it was you).
I was also wandering what exactly was the intention here. I'm a bit concerned that the medaka-consensus is used as 
input for this rule and not the unbiased reference (which I think should be the normal use).